testdata/numgen.js

```diff
@@ -0,0 +1,117 @@
+	0x3f0179ec9cbd821en, 0x3ecbf647612f3696n, 0x3e965e9f80f29212n, 0x3e61e54c672874dbn,
+	0x3e2ca213d840baf8n, 0x3df6e80fe033c8c6n, 0x3dc2533fe68fd3d2n, 0x3d8d51ffd74c861cn,
+	0x3d5774ccac3d3817n, 0x3d22c3d6f030f9acn, 0x3cee0624b3818f79n, 0x3cb804ea293472c7n,
+	0x3c833721ba905bd3n, 0x3c4ebe9c5db3c61en, 0x3c18987d17c304e5n, 0x3be3ad30dfcf371dn,
+	0x3baf7b816618582fn, 0x3b792f9ab81379bfn, 0x3b442615600f9499n, 0x3b101e77800c76e1n,
+	0x3ad9ca58cce0be35n, 0x3aa4a1e0a3e6fe90n, 0x3a708180831f320dn, 0x3a3a68cd9e985016n,
+	0x4024000000000000n, 0x4014000000000000n, 0x3fe0000000000000n, 0x3fa999999999999an,
+	0x3f747ae147ae147bn, 0x3f40624dd2f1a9fcn, 0x3f0a36e2eb1c432dn, 0x3ed4f8b588e368f1n,
+	0x3ea0c6f7a0b5ed8dn, 0x3e6ad7f29abcaf48n, 0x3e35798ee2308c3an, 0x3ed539223589fa95n,
+	0x3ed4ff26cd5a7781n, 0x3ed4f95a762283ffn, 0x3ed4f8c60703520cn, 0x3ed4f8b72f19cd0dn,
+	0x3ed4f8b5b31c0c8dn, 0x3ed4f8b58d1c461an, 0x3ed4f8b5894f7f0en, 0x3ed4f8b588ee37f3n,
+	0x3ed4f8b588e47da4n, 0x3ed4f8b588e3849cn, 0x3ed4f8b588e36bb5n, 0x3ed4f8b588e36937n,
+	0x3ed4f8b588e368f8n, 0x3ed4f8b588e368f1n, 0x3ff0000000000000n, 0xbff0000000000000n,
+	0xbfeffffffffffffan, 0xbfeffffffffffffbn, 0x3feffffffffffffan, 0x3feffffffffffffbn,
+	0x3feffffffffffffcn, 0x3feffffffffffffen, 0xbfefffffffffffffn, 0xbfefffffffffffffn,
+	0x3fefffffffffffffn, 0x3fefffffffffffffn, 0x3fd3333333333332n, 0x3fd3333333333333n,
+	0x3fd3333333333334n, 0x0010000000000000n, 0x000ffffffffffffdn, 0x000fffffffffffffn,
+	0x7fefffffffffffffn, 0xffefffffffffffffn, 0x4340000000000000n, 0xc340000000000000n,
+	0x4430000000000000n, 0x44b52d02c7e14af5n, 0x44b52d02c7e14af6n, 0x44b52d02c7e14af7n,
+	0x444b1ae4d6e2ef4en, 0x444b1ae4d6e2ef4fn, 0x444b1ae4d6e2ef50n, 0x3eb0c6f7a0b5ed8cn,
+	0x3eb0c6f7a0b5ed8dn, 0x41b3de4355555553n, 0x41b3de4355555554n, 0x41b3de4355555555n,
+	0x41b3de4355555556n, 0x41b3de4355555557n, 0xbecbf647612f3696n, 0x43143ff3c1cb0959n,
+])
+staticF64s = new Float64Array(staticU64s.buffer)
+serialU64s = new BigUint64Array(2000)
+for (i = 0; i < 2000; i++) {
+	serialU64s[i] = 0x0010000000000000n + BigInt(i)
+}
+serialF64s = new Float64Array(serialU64s.buffer)
+
+var state = {
+    idx: 0,
+    data: new Float64Array(),
+    block: new ArrayBuffer(32),
+}
+function next() {
+    var f = 0.0;
+    if (state.idx < staticF64s.length) {
+        f = staticF64s[state.idx]
+    } else if (state.idx < staticF64s.length + serialF64s.length) {
+        f = serialF64s[state.idx - staticF64s.length]
+    } else {
+        while (f == 0.0 || !isFinite(f)) {
+            if (state.data.length == 0) {
+                state.block = crypto.createHash("sha256").update(new Buffer(state.block)).digest().buffer
+                state.data = new Float64Array(state.block)
+            }
+            f = state.data[0]
+            state.data = state.data.slice(1)
+        }
+    }
+    state.idx++
+    return f
+}
+
+// TODO: Emit the file as GZIP compressed.
+// TODO: Fix a buffering issue where Node.js tries to buffer the entire output.
```

Comment:
I don't know node.js well enough to know why it tries to buffer the entire output file even though this is a `WriteStream`.

Regardless of the poor performance, this does correctly produce the file given enough RAM.